### ISSUE TITLE
Logic of SAVE progress seems wield

### ISSUE THREAD

**Issue by Stupidism** [guns2410/redux-storage#2]
# Use Case
I wanted to add a middleware to modify payload to sync into storage-engine, but failed all the time.

# Problem
Turnen out newState is [saved into storage-engine before its action dispatched](https://github.com/react-stack/redux-storage/blob/master/src/createMiddleware.js#L97).

I think the right order should be:
1. dispatch this SAVE action 
2. save into storage-engine in reducer or middleware.

**Comment by guns2410** [comment]
What do you mean to modify payload to sync into storage-engine? Does this mean that you need your redux store to have different data than what is stored in the storage-engine?

**Comment by Stupidism** [comment]
No, it means I want to add redux middleware for `SAVE` action and modify the payload in the middleware. But failed because of the logic sequence.
